Snowflake.Data.Tests/SFConnectionIT.cs

```diff
@@ -299,16 +299,36 @@ public void TestUnknownAuthenticator()
         [Test]
         [Ignore("This test requires manual setup and therefore cannot be run in CI")]
         public void TestOktaConnection()
+        {
+            using (var conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString 
+                    = ConnectionStringWithoutAuth
+                    + String.Format(
+                        ";authenticator={0};user={1};password={2};",
+                        testConfig.OktaURL,
+                        testConfig.OktaUser,
+                        testConfig.OktaPassword);
+                conn.Open();
+                Assert.AreEqual(ConnectionState.Open, conn.State);
+            }
+        }
+
+        [Test]
+        [Ignore("This test requires manual interaction and therefore cannot be run in CI")]
+        public void TestSSOConnection()
         {
             using (IDbConnection conn = new SnowflakeDbConnection())
             {
-                conn.ConnectionString = "scheme=http;host=testaccount.reg.snowflakecomputing.com;port=8082;user=qa@snowflakecomputing.com;password=Test123!;" +
-                    "account=testaccount;role=sysadmin;db=testdb;schema=public;warehouse=regress;authenticator=https://snowflakecomputing.okta.com";
+                conn.ConnectionString
+                    = ConnectionStringWithoutAuth
+                    + ";authenticator=externalbrowser;";

```

Comment:
I realized that you don't even need to have a user specified to have externalbrowser authentication. What are you guys' thought? My feeling as a user is that this is a little cumbersome given that I should type in my user name in the browser and also have user specified in the connection string;

Reply:
User is required to match the identity. In particular, token id cache will require to match.

Reply:
See this: https://github.com/snowflakedb/snowflake-connector-python/blob/master/connection.py#L756
oAuth won't take USER.

Reply:
More accurately LOGIN_NAME is required. So LOGIN_NAME is not necessary the same as SSO user name.
NOTE: Snowflake drivers take LOGIN_NAME as `USER`.

Reply:
Addressed